sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -275,36 +206,49 @@ public void populateDisplayData(DisplayData.Builder builder) {
             .add(DisplayData.item("inputMetadata", sb.toString())
             .withLabel("Input Metadata"));
       }
-      if (Boolean.TRUE.equals(getParseSynchronously())) {
-        builder
-          .add(DisplayData.item("parseMode", "synchronous")
-            .withLabel("Parse Mode"));
-      } else {
-        builder
-          .add(DisplayData.item("parseMode", "asynchronous")
-            .withLabel("Parse Mode"));
-        builder
-          .add(DisplayData.item("queuePollTime", getQueuePollTime().toString())
-            .withLabel("Queue Poll Time"))
-        .add(DisplayData.item("queueMaxPollTime", getQueueMaxPollTime().toString())
-          .withLabel("Queue Max Poll Time"));
-      }
-      Integer minTextLen = getMinimumTextLength();
-      if (minTextLen != null && minTextLen > 0) {
-        builder
-        .add(DisplayData.item("minTextLen", getMinimumTextLength().toString())
-          .withLabel("Minimum Text Length"));
-      }
-      if (Boolean.TRUE.equals(getReadOutputMetadata())) {
-        builder
-          .add(DisplayData.item("readOutputMetadata", "true")
-            .withLabel("Read Output Metadata"));
-      }
     }
 
-    @Override
-    protected Coder<String> getDefaultOutputCoder() {
-      return StringUtf8Coder.of();
+    private static class ParseToStringFn extends DoFn<ReadableFile, ParseResult> {
+
+      private static final long serialVersionUID = 6837207505313720989L;
+      private TikaIO.ParseAll spec;
+      ParseToStringFn(TikaIO.ParseAll spec) {
+        this.spec = spec;
+      }
+      @ProcessElement
+      public void processElement(ProcessContext c) throws IOException {
+        ReadableFile file = c.element();
+        InputStream stream = Channels.newInputStream(file.open());
+
+        final InputStream is = TikaInputStream.get(stream);
+        TikaConfig tikaConfig = null;
+        if (spec.getTikaConfigPath() != null) {
+          try {
+            tikaConfig = new TikaConfig(spec.getTikaConfigPath().get());
+          } catch (TikaException | SAXException e) {
+            throw new IOException(e);
+          }
+        }
+        final Parser parser = tikaConfig == null ? new AutoDetectParser()
+            : new AutoDetectParser(tikaConfig);
+        final ParseContext context = new ParseContext();
+        context.set(Parser.class, parser);
+        org.apache.tika.metadata.Metadata tikaMetadata = spec.getInputMetadata() != null
+          ? spec.getInputMetadata() : new org.apache.tika.metadata.Metadata();
+
+        ContentHandler tikaHandler = new ToTextContentHandler();
```

Comment:
Can a lot of this be abbreviated to Tika.parseToString(is) ?

Reply:
I've reviewed that code, IMHO we should retain the control in TikaIO, that function uses
https://github.com/apache/tika/blob/master/tika-core/src/main/java/org/apache/tika/sax/BodyContentHandler.java

which is apparently optimized for reading the XML content, and the goal was also to let users provide custom content handlers.

Tim also recommended the other day supporting the recursive ParserWrapper,
https://github.com/apache/tika/blob/master/tika-example/src/main/java/org/apache/tika/example/ParsingExample.java#L138
so may be having a Metadata only (in ParseResult will not be enough), I suppose we can still keep ParseResult.getMetadata (the 1st one in the list) but add later on ParseResult.getEmbeddedMetadata too, once we enhance the code to use the recursive parser